nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/model/impl/TestArrayProperty.java

```diff
@@ -47,4 +59,64 @@ public void testCollectionOfIntOnLongProperty() {
         assertArrayEquals(new Long[] { 1L, 2L, 3L }, (Long[]) property.getValue());
     }
 
+    @Test
+    public void testArrayOfStrings() {
+        ArrayProperty property = getArrayProperty(StringType.INSTANCE);
+        String[] testStrings = new String[] {"there", null, "once","was a dog"};
+        property.setValue(testStrings);
+        assertArrayEquals(testStrings, (String[]) property.getValue());
+    }
+
+    @Test
+    public void testArrayOfDoubles() {
+        ArrayProperty property = getArrayProperty(DoubleType.INSTANCE);
+
+        property.setValue(new String[] {"1","2","3"});
+        assertArrayEquals(new Double[] { 1d, 2d, 3d }, (Double[]) property.getValue());
+
+        property.setValue(new Object[] {"1",2,"3", 9,  null});
+        assertArrayEquals(new Double[] { 1d, 2d, 3d, 9d, null}, (Double[]) property.getValue());
+
+        property.setValue(Arrays.asList(1f, 5f, 123.4f));
+        assertArrayEquals(new Double[] { 1d, 5d, (double)123.4f}, (Double[]) property.getValue());
+    }
+
+    @Test
+    public void testArrayOfIntegers() {
+        ArrayProperty property = getArrayProperty(IntegerType.INSTANCE);
+
+        property.setValue(new String[] {"1","2", null,"3"});
+        assertArrayEquals(new Long[] { 1L, 2L, null, 3L }, (Long[]) property.getValue());
+
+        property.setValue(Arrays.asList(1, "2", 3));
+        assertArrayEquals(new Long[] { 1L, 2L, 3L }, (Long[]) property.getValue());
+    }
+
+    @Test
+    public void testArrayOfDates() throws ParseException {
```

Comment:
You can just `'throws Exception` in unit tests, as we don't really care about the specific type.